### PR TITLE
Privacy check bug

### LINKED ISSUE #153: Privacy Check Bugs
Privacy is not being checked for viewing friend kanbans or sending cover letter requests.

### PR DESCRIPTION
**Pull request by LokiFrostGiant**
## Tickets
Closes #153 
## Summary Of Changes
Fixes bug where review requests and kanban viewing could be done despite privacy settings.
## Notes for Reviewers
All tests except acceptance tests implemented.

### CODE REVIEW

**Comment by drkitt** [comment]
The thing I pointed out at the end of class (preventing board viewing and review requesting worked perfectly, but allowing them again required a refresh) isn't happening anymore. Maybe the refresh I did after showing you the bug had some deeper effect? In any case, it seems to work perfectly in all cases. I'll check it out again when the tests are ready, but it looks good for now.

File: frontend/src/components/modal/job/JobDetailModal.vue [hunk 1]
```diff
@@ -185,7 +193,11 @@
             id="job-save-button"
             color="blue-darken-1"
             variant="text"
-            @click="this.saveClicked"
+            @click="
```
Comment (jaredwebber) [inline frontend/src/components/modal/job/JobDetailModal.vue:196]: It looks like job creation is broken on this branch - nothing happens when I click save when trying to add a job 🤔

Comment (LokiFrostGiant) [reply frontend/src/components/modal/job/JobDetailModal.vue:196]: Simply forgot to add `()` when adding more to the `@click` effect. Resolved.

File: frontend/src/components/modal/job/ReviewRequestModal.vue [hunk 2]
```diff
@@ -114,6 +114,15 @@ export default {
...
+    }
+  },
+  beforeMount() {
+    if (
+      this.activeUser.friends !== null &&
+      this.activeUser.friends !== undefined
+    ) {
+      this.checkSendability()
     }
   },
```
Comment (jaredwebber) [inline frontend/src/components/modal/job/ReviewRequestModal.vue:127]:
```suggestion
  mounted() {
    this.checkSendability()
  },
```

File: frontend/src/components/modal/job/ReviewRequestModal.vue [hunk 3]
```diff
@@ -142,7 +151,17 @@ export default {
         })
       })
     },
-
+    async checkSendability() {
+      this.activeUser.friends.forEach(async (friend) => {
```
Comment (jaredwebber) [inline frontend/src/components/modal/job/ReviewRequestModal.vue:155]:
```suggestion
      this.activeUser.friends?.forEach(async (friend) => {
```

**Review by jaredwebber — COMMENTED** [review]
Left a couple suggestions you could try out (they pass tests for me, didnt test functionality). Unless theres a reason to use `beforeMount`, might be easier to just use `mounted` as we've used it elsewhere in the app, and the `?` operator is a bit more conscise than the explicit `!= null && != undef` check.

**Review by jaredwebber — COMMENTED** [review]
Appears to work properly, curious if @Speuce knows if theres a better way to await the tests than the way we did it.

Not really part of this ticket, but if you could change the cover letter review request message to say something along the lines of "No friends available to review", it might be better UI, since I *do* have friends, just they dont allow cover letter review requests.
Screenshot 2023-03-28 at 12 00 17 PM

**Comment by LokiFrostGiant** [comment]
> Appears to work properly, curious if @Speuce knows if theres a better way to await the tests than the way we did it.
> 
> Not really part of this ticket, but if you could change the cover letter review request message to say something along the lines of "No friends available to review", it might be better UI, since I _do_ have friends, just they dont allow cover letter review requests. 

Change made and resolved.

**Pull request merged**